flocx_market/db/sqlalchemy/api.py

```diff
@@ -249,26 +249,33 @@ def contract_destroy(contract_id, context):
 
 # offer_contract_relationship
 def offer_contract_relationship_get(context,
-                                    contract_id=None,
-                                    marketplace_offer_id=None):
+                                    offer_contract_relationship_id=None):
 
-    if (contract_id is not None) and (marketplace_offer_id is None):
-        return get_session().query(models.OfferContractRelationship).filter_by(
-            contract_id=contract_id).all()
-
-    elif (contract_id is None) and (marketplace_offer_id is not None):
-        return get_session().query(models.OfferContractRelationship).filter_by(
-            marketplace_offer_id=marketplace_offer_id).all()
-    elif (contract_id is not None) and (marketplace_offer_id is not None):
-        return get_session().query(models.OfferContractRelationship) \
-            .filter(models.OfferContractRelationship.contract_id
-                    == contract_id,
-                    models.OfferContractRelationship.marketplace_offer_id
-                    == marketplace_offer_id).first()
+    if offer_contract_relationship_id is None:
+        return None
+    return get_session().query(models.OfferContractRelationship).filter(
+        models.OfferContractRelationship.offer_contract_relationship_id
+        == offer_contract_relationship_id).first()
 
 
-def offer_contract_relationship_get_all(context):
-    return get_session().query(models.OfferContractRelationship).all()
+def offer_contract_relationship_get_all(context, filters=None):
+    if filters is None:
+        return get_session().query(models.OfferContractRelationship).all()
+    else:
+        query = []
+        for field in ['marketplace_offer_id', 'contract_id', 'status']:
+            if field in filters:
+                query = get_session().query(models.OfferContractRelationship)\
+                    .filter_by(**{field: filters[field]})
```

Comment:
Try something like this instead:

```
    query = get_session().query(models.OfferContractRelationship)
    for field in ['marketplace_offer_id', 'contract_id', 'status']:
            if field in filters:
                query = query.filter_by(**{field: filters[field]})
    return query.all()
```

Let me know if that works!

Reply:
Oh, and theoretically this should eliminate the need for lines 271-277 below.

Reply:
it works for the single argument queries, but the multiple arg queries are failing with it

Reply:
Can I see the code you're using for the multiple arg queries that are failing? It worked for me with offers, so I'm confused why it'd be failing here.

Reply:
the tests are failing @tzumainn

Reply:
Right - what code are you using when the test is failing? Is it just this code, in which case I'd suggest using my modified version, or something else?

Reply:
yeah just this code, I run tox and it won't pass

Reply:
In the, /db/sqlalchemy/test_api.py the ocr tests are failing

Reply:
Can you try the version of the code similar to what I pasted above? You set `query` *outside* of the `for` loop, and then add filters one-by-one.

Reply:
`def offer_contract_relationship_get_all(context, filters={}):
    if len(filters) == 0:
        return get_session().query(models.OfferContractRelationship).all()
    else:
        query = get_session().query(models.OfferContractRelationship)
        for field in ['marketplace_offer_id', 'contract_id', 'status']:
            if field in filters:
                query = query.filter_by(**{field: filters[field]})
        return query.all()`

Reply:
yeah that's what I did, I just replaced the bottom part of the code with what you pasted

Reply:
<img width="583" alt="Screen Shot 2019-08-01 at 11 11 52 AM" src="https://user-images.githubusercontent.com/25260955/62305117-4daa5200-b44d-11e9-87d9-c343bd94d153.png">

Reply:
okay - what tests are failing at that point?

Reply:
In the, /db/sqlalchemy/test_api.py the ocr tests are failing.
The problem is that when we are querying by multiple parameters and not just the uuid, the query is not getting chained on and is only returning an empty list

Reply:
Okay, I took a look, and the tests worked for me with my suggested changes as soon as I updated a few things. Instead of:

```
def test_offer_contract_relationship_create_valid(app, db, session):
    contract_data, offer_test_id = create_test_contract_data_for_ocr()
    contract = api.contract_create(contract_data, admin_context)
    filters = {
	'marketplace_offer_id': offer_test_id,
        'contract_id': contract.contract_id
    }
    ocr = api.offer_contract_relationship_get_all(
        context=admin_context,
        filters=filters,
    )
    assert contract.contract_id == ocr.contract_id
```

I changed the last line to

```
   assert contract.contract_id == ocr[0].contract_id
```

Reply:
that is so weird ... I changed this line `ocr = api.offer_contract_relationship_get_all(
        context=admin_context,
        filters=filters,
    )`

by adding to the end [0] and I was getting an error because it was an empty list...
Which is basically the same thing. I'm going to try your version now

Reply:
and of course, it works now :D love when stuff like this happens. Pushed the updates @tzumainn